x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -44,11 +44,13 @@ import (
 )
 
 const (
-	maxRetriesstoreAgentInfo = 5
-	waitingForAgent          = "Waiting for Elastic Agent to start"
-	waitingForFleetServer    = "Waiting for Elastic Agent to start Fleet Server"
-	defaultFleetServerHost   = "0.0.0.0"
-	defaultFleetServerPort   = 8220
+	maxRetriesstoreAgentInfo       = 5
+	waitingForAgent                = "Waiting for Elastic Agent to start"
+	waitingForFleetServer          = "Waiting for Elastic Agent to start Fleet Server"
+	defaultFleetServerHost         = "0.0.0.0"
+	defaultFleetServerPort         = 8220
+	defaultFleetServerInternalHost = "localhost"
+	defaultFleetServerInternalPort = 8221
```

Comment:
Lets make sure we document this internal as needed by fleet-server. @jlind23 Can you make sure we follow up on this? We also need to have this in the changelog as if by chance someone uses this port already on the machine, it would break things (hopefully unlikely).

Reply:
@ruflin Where should it be documented though?

Reply:
Two ideas:

* https://www.elastic.co/guide/en/fleet/current/fleet-faq.html 
* https://www.elastic.co/guide/en/fleet/master/fleet-troubleshooting.html

@bmorelli25 Where would these internal docs around elastic-agent and fleet-server fit best?

Reply:
This is @dedemorton's territory. DeDe, can you take a look?

Reply:
OK, so it sounds like you don't want to document the internal port setting in the main docs (command reference + Fleet Server settings) because users will not set it in a production environment. It also sounds like we're covered for the situation where the port is already in use (the code will select an unused port when there's a conflict).

I'd like to understand specific use cases for this setting, though, because I'm always suspicious of hidden settings that we don't want to document for users. We have a lot of users who are on the support team and might look in the reference first. My preference is to have reference docs that are complete, but to make the usage clear in the description.

Sounds like it also makes sense to cover this setting in the [troubleshooting](https://www.elastic.co/guide/en/fleet/master/fleet-troubleshooting.html) docs, but we need to explain when/why to change this setting during testing.

We could also mention in the Fleet Server docs that Fleet Server uses port 8221 internally, but will look for another port if that one isn't available.

Reply:
It will not look for antoher port. We might want to use the port for testing purpose or as an escape hatch if indeed a user has a conflict. 

There might we 2 things we should document:

* Architecture / Internals: What port is used for internal communication -> public
* Hidden config to change it -> developer focus, mentioned that not officially supported